### PR TITLE
Github Action for building and testing PRs

### PR DESCRIPTION
**Pull request by JonathanDCohen**

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Build and Test
+
+on: push
+
+jobs:
+  build-and-test:
+    name: Build and Test Web App
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (fortuna) [inline .github/workflows/build_and_test.yml:10]: You don't seem to need a matrix.
Move the steps to the steps list.

Comment (JonathanDCohen) [reply .github/workflows/build_and_test.yml]: Done in 6ccdf52

**Comment by JonathanDCohen** [comment]
I've also change the repo branch protectionsto require the Github Action instead of the Travis build

**Review by fortuna — APPROVED** [review]
Thanks for doing this!

**Pull request merged**